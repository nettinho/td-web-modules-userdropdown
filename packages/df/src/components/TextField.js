import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";

export const TextField = ({ label, name, required, value, onChange }) => (
  <Form.TextArea
    label={label}
    required={required}
    value={value}
    name={name}
    onChange={onChange}
  />
);

TextField.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default TextField;
