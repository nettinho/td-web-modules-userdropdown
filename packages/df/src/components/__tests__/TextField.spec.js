import React from "react";
import { shallow } from "enzyme";
import { TextField } from "../TextField";

describe("<TextField />", () => {
  const label = "label";
  const name = "name";
  const required = true;
  const value = "Foo";
  const onChange = jest.fn();
  const props = { label, name, required, value, onChange };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<TextField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
