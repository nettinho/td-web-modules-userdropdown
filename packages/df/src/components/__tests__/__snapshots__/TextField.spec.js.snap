// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> matches the latest snapshot 1`] = `
<FormTextArea
  as={[Function]}
  control={[Function]}
  label="label"
  name="name"
  onChange={[MockFunction]}
  required={true}
  value="Foo"
/>
`;
